glean-core/ios/Glean/Glean.swift

```diff
@@ -104,10 +114,29 @@ public class Glean {
     /// - parameters:
     ///     * enabled: When true, enable metric collection.
     public func setUploadEnabled(_ enabled: Bool) {
-        uploadEnabled = enabled
-
         if isInitialized() {
-            glean_set_upload_enabled(handle, enabled.toByte())
+            let originalEnabled = getUploadEnabled()
+
+            _ = Dispatchers.shared.launch {
```

Comment:
Is this something that we want to execute on the serial queue?  If we don't want to worry about what else is in the Glean Dispatchers, you can also use Grand Central Dispatch to execute this on the fly on a background thread like this:
```Swift
DispatchQueue.global().async {
	// Background work
}
```

Reply:
Right now, yes, as it then runs for sure after any uploader.
Once the other code lands we should re-evaluate where this runs?